### PR TITLE
Added statements to OLC (Expression, Print, Var, Assignment)

### PR DESCRIPTION
**Pull request by gbarbachov**
Added:
1) ExpressionStatement to parser and interpreter
2) PrintStatement to parser and interpreter
3) VarStatement to parser
4) Variable expression to parser
5) Variable statement and expression to the Interpreter (global scope only)
6) Added Assignment to both parser and interpreter

This required a lot of changes, especially for the tests. Some AST tests are still failing, commented out for now.

### CODE REVIEW

File: olc [hunk 1]
```diff
@@ -4,23 +4,32 @@ import sys
...
+        lexer.process_source_code()
+        tokens = lexer.get_tokens()
+
+        parser = Parser(tokens)
+        statements = parser.parse()
+
+        # if statements:
+        interpreter = Interpreter()
+        try:
+            interpreter.interpret(statements)
```
Comment (togna) [inline olc:26]: With this change the orchestrator is going to create a new lexer, parser, and interpreter for every call to execute. Do you mind helping me understand the purpose for this?

File: olc [hunk 2]
```diff
@@ -32,9 +41,7 @@ class Orchestrator:
                 line = input()
                 if line is None:
                     break
-                result = self.execute(line)
-                if result:
-                    print(result)
+                self.execute(line)
```
Comment (togna) [inline olc:44]: With this change I don't think the output will be printed to the terminal when running the repl, but I think that it should be printed. Can you let me know the intent of this change

File: src/parser.py [hunk 3]
```diff
@@ -1,38 +1,114 @@
...
+    UNARY_TOKENS, LITERAL_TOKENS, IGNORED_TOKENS, GROUP_OPENING_TOKENS, GROUP_CLOSING_TOKENS, \
+    STATEMENT_START_TOKENS, STATEMENT_END_TOKENS, IDENTIFIER_TOKENS, EQUALS_TOKENS, \
+    VAR_STATEMENT_TOKENS, \
+    PRINT_STATEMENT_TOKENS
 
 
 class Parser:
     def __init__(self, tokens=None):
-        self.tokens = [token for token in tokens if token not in IGNORED_TOKENS] if tokens else []
+        self.tokens = [token for token in tokens if token.token_type not in IGNORED_TOKENS] if tokens else []
```
Comment (togna) [inline src/parser.py:18]: Why can't we just do self.tokens = tokens here and set the tokens default value to an empty list?

**Review by togna — COMMENTED** [review]
There is a lot implemented here which is great, and the implementation logically makes sense. I see you changed a couple integration tests, although it doesn't seem like any of the new functionality is tested right? That's making me a little less confident about this. I'm also a little concerned about the commented tests. Can you fix those before we merge and add at least some basic tests for the new functionality? Besides that I just had a few questions about the changes in the orchestrator class. You've made a ton of progress this week!

**Review by gaidosj — APPROVED** [review]
Discussed. Approved.

**Pull request merged**